### PR TITLE
Write tests for the link checker and refactor

### PR DESCRIPTION
**Pull request by laurentqro**
To be able to display accurate and trustworthy data for https://github.com/alphagov/datagovuk_publish/pull/505, we need to make sure the link checker is doing the job.

This PR adds tests for the link checker, and a refactoring.

### CODE REVIEW

File: spec/services/link_checker_service_spec.rb [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+      link = FactoryGirl.create(:link, url: "http://www.brokenlink.com")
+      service = described_class.new(link)
+
+      stub_request(:head, link.url).to_return(status: 404)
+
+      service.run
+
+      task = Task.last
+
+      expect(task.organisation_id).to eql link.dataset.organisation_id
```
Comment (andrewgarner) [inline spec/services/link_checker_service_spec.rb]: How about this removing some of the repetition by using the [have_attributes](https://relishapp.com/rspec/rspec-expectations/v/3-7/docs/built-in-matchers/have-attributes-matcher) matcher?

```ruby
expect(Task.last)
  .to have_attributes(
        organisation_id: link.dataset.organisation_id,
        owning_organisation: link.dataset.organisation.name,
        required_permission_name: be_blank,
        category: "broken",
        quantity: 1,
        related_object_id: link.dataset.uuid,
        description: %('#{link.dataset.title}' contains broken links)
      )
```

Comment (laurentqro) [reply spec/services/link_checker_service_spec.rb]: oooh nice

File: app/services/link_checker_service.rb [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  def last_modified
+    timestamp = response.headers[:last_modified]
+    return Time.parse(timestamp) if !timestamp.nil?
+  end
+
+  def file_size
+    response.headers[:content_length].to_i
+  end
+
+  def working?
```
Comment (andrewgarner) [inline app/services/link_checker_service.rb]: Ruby's Net::HTTP library has this behaviour built in. How about we defer to that?

```ruby
response.net_http_res.is_a?(Net::HTTPSuccess)
```

Comment (laurentqro) [reply app/services/link_checker_service.rb]: Implemented in https://github.com/alphagov/datagovuk_publish/pull/517

**Pull request merged**